compiler/rustc_trait_selection/src/error_reporting/infer/need_type_info.rs

```diff
@@ -613,6 +666,130 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
             }),
         }
     }
+
+    fn get_suggestions(
+        &self,
+        self_ty: Ty<'tcx>,
+        def_id: DefId,
+        target_type: bool,
+        param_env: ty::ParamEnv<'tcx>,
+        args: Option<&ty::GenericArgs<'tcx>>,
+        predicate: Option<ty::Predicate<'tcx>>,
```

Comment:
An arg named `predicate` that actually only cares about a single predicate kind is probably best passed as just that single predicate kind :thinking: and we should probably give it a better name.

Reply:
I wanted to leave the filtering from the `Predicate` in this method so as not to complicate the code in the caller twice-removed (as they don't care what the `Predicate` actually is, we're the ones we care if this is a `<_ as Mul>::Output = _`.

Reply:
Changed to pass in only the `ProjectionPredicate` and renamed to `originating_projection` to try to make it clearer what this is (happy to hear better options).